### PR TITLE
Functions

### PR DESCRIPTION
**Pull request by ElenaLadutska**
Functions tasks completed

### CODE REVIEW

File: functions.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
+//Task 1. Напишите функцию, которая принимает 2 числа и возвращает 1, если первое число больше, чем второе; 
+//-1, если первое число меньше, чем второе, и 0, если числа равны.
+//Попробуйте реализовать эту функцию не только с помощью if...else, 
+//но и с помощью тернарного оператора, и в виде стрелочной функции.
+//const num1 = +prompt('Enter a number');
+//const num2 = +prompt('Enter one more number');
+
+//function numberComparison(num1, num2) {
+//  if (num1 > num2) {
```
Comment (kathrynowy) [inline functions.js]: where is check for NaN?

File: functions.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
+//Task 1. Напишите функцию, которая принимает 2 числа и возвращает 1, если первое число больше, чем второе; 
+//-1, если первое число меньше, чем второе, и 0, если числа равны.
+//Попробуйте реализовать эту функцию не только с помощью if...else, 
+//но и с помощью тернарного оператора, и в виде стрелочной функции.
+//const num1 = +prompt('Enter a number');
+//const num2 = +prompt('Enter one more number');
+
+//function numberComparison(num1, num2) {
```
Comment (kathrynowy) [inline functions.js]: if you already have read about arrow functions - use them

File: functions.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+//  if (num1 > num2) {
+//    return '1';
+//  }else if (num1 < num2) {
+//    return '-1';
+//  }else {
```
Comment (kathrynowy) [inline functions.js]: you can remove last if for a better readibility

File: functions.js [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+//    return '0';
+//  };
+//};
+//alert(numberComparison(num1, num2));
+
+//function numberComparison(num1, num2) {
+//  (num1 > num2) ? alert('1'):
+//  (num1 < num2) ? alert('-1'):
+//  alert('0');
```
Comment (kathrynowy) [inline functions.js]: it's better not to use double /triple etc. ternaries

File: functions.js [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+//     return `Result of multiplication:${num1 * num2}`;
+//   }else {
+//     if(num1 % 2){
+//       return `Odd is ${num1}`;
+//   }else {
+//     return `Odd is ${num2}`
+//   };
+//   };
+// };
+//console.log(isEven(num1,num2));
```
Comment (kathrynowy) [inline functions.js]:
```suggestion
// function isEven(num1, num2) {
const  isFirstNumEven = num1 % 2;
const  isSecondNumEven = num1 % 2;

if (!isFirstNumEven && !isSecondNumEven) {
 return `Result of multiplication:${num1 * num2}`;
}

if (isFirstNumEven && isFirstNumEven) {
 return `Sum:${num1 + num2}`;
}

  return  `Odd is ${isFirstNumEven ? num1: num2}`;
  
//console.log(isEven(num1,num2));
```

Comment (kathrynowy) [reply functions.js]: обрати внимание, как я избавилась от вложенности + вынесла проверку в переменную , чтобы не дублировать код

последний ретурн (при не попадании ни в один иф выполнится итак)

File: functions.js [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+//Написать функцию, которая принимает 1 аргумент (возраст пользователя)
+//и возвращает boolean значение - можно ему на сайт или нет.( если 18 и больше лет)
+// const age = +prompt('Enter your age');
+
+// function isAdult(age) {
+//   if (age >= 18) {
+//     return true;
+//   }else {
+//     return false;
+//   };
```
Comment (kathrynowy) [inline functions.js]: in 1 line with 1 return only

File: functions.js [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+ 
+//Task 5
+//Написать функцию, которая принимает 3 аргумента,
+//если 1 аргумент true - то складывать 2 и 3 аргумент , если 1-ый аргумент false  - вычитать 2 и 3 аргументы
+//function isFirstArgumentTrue(first, second, third) {
+//  if (first) {
+//    return second + third;
+//  } else {
+//    return second - third;
+//  }
```
Comment (kathrynowy) [inline functions.js]: same

File: functions.js [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+//   }else {
+//     return false;
+//   };
+// };
+// console.log(isAdult(age));
+ 
+//Task 5
+//Написать функцию, которая принимает 3 аргумента,
+//если 1 аргумент true - то складывать 2 и 3 аргумент , если 1-ый аргумент false  - вычитать 2 и 3 аргументы
+//function isFirstArgumentTrue(first, second, third) {
```
Comment (kathrynowy) [inline functions.js]: we name functions only with verbs in the beginning

File: functions.js [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+ 
+//Task 6
+//Написать функцию, которая принимает 1 аргумент (кол-во итераций цикла)
+//и выводит на страницу столько красных кругов на страницу
+//const amountOfIterations = +prompt('Enter a number of iterations');
+
+// function redCircleDrawing(amountOfIterations) {
+//   for (let i = 1; i <= amountOfIterations; i++) {
+//     document.open();
+//     document.write(`<div class='red-circles'></div>`);
```
Comment (kathrynowy) [inline functions.js]: create Element

**Review by kathrynowy — CHANGES_REQUESTED** [review]

File: functions.js (hunk 1 above)
Comment (ElenaLadutska) [reply functions.js]: fixed

File: functions.js (hunk 2 above)
Comment (ElenaLadutska) [reply functions.js:8]: From line 28 the arrow function in new commit

File: functions.js (hunk 4 above)
Comment (ElenaLadutska) [reply functions.js]: okay

File: functions.js (hunk 3 above)
Comment (ElenaLadutska) [reply functions.js:13]: why? the task specifies that if the numbers are equal return 0

File: functions.js (hunk 5 above)
Comment (ElenaLadutska) [reply functions.js]: fixed

File: functions.js (hunk 3 above)
Comment (ElenaLadutska) [reply functions.js]: fixed

File: functions.js (hunk 6 above)
Comment (ElenaLadutska) [reply functions.js]: fixed

File: functions.js (hunk 7 above)
Comment (ElenaLadutska) [reply functions.js]: fixed

File: functions.js (hunk 9 above)
Comment (ElenaLadutska) [reply functions.js]: fixed

File: functions.js (hunk 8 above)
Comment (ElenaLadutska) [reply functions.js]: okay, I renamed the function

File: functions.css [hunk 10]
```diff
@@ -0,0 +1,11 @@
+:root {
+  --radical-red: #FF3F58;
+}
+.red-circles {
+  width: 20px;
```
Comment (kathrynowy) [inline functions.css]:
```suggestion
}

.red-circles {
  width: 20px;
```

File: functions.js [hunk 11]
```diff
@@ -0,0 +1,134 @@
...
+//-1, если первое число меньше, чем второе, и 0, если числа равны.
+//Попробуйте реализовать эту функцию не только с помощью if...else, 
+//но и с помощью тернарного оператора, и в виде стрелочной функции.
+// const num1 = +prompt('Enter a number');
+// const num2 = +prompt('Enter one more number');
+
+// function numberComparison(num1, num2) {
+//   if (num1 > num2) {
+//     return 1;
+//   }else if (num1 < num2) {
```
Comment (kathrynowy) [inline functions.js]:
```suggestion
//   } else if (num1 < num2) {
```

File: functions.js [hunk 12]
```diff
@@ -0,0 +1,134 @@
...
+//     return -1;
+//   }else if (isNaN(num1) || isNaN(num2)){
+//     return 'You entered a text. Enter a number, pls'
+//   }
+//     return 0;
```
Comment (kathrynowy) [inline functions.js:19]:
```suggestion

//     return 0;
```

Comment (kathrynowy) [reply functions.js:19]: empty line before return if it's not on the 1 line

File: functions.js [hunk 13]
```diff
@@ -0,0 +1,134 @@
...
+//function numberComparison(num1, num2) {
+//  (num1 > num2) ? alert('1'):
+//  (num1 < num2) ? alert('-1'):
+//  alert('0');
+//};
+//numberComparison(num1, num2);
+
+// const numberComparison = (num1, num2) => {
+//  if (num1 > num2) {
+//    return '1';
```
Comment (kathrynowy) [inline functions.js]: why is it like text?

File: functions.js [hunk 14]
```diff
@@ -0,0 +1,134 @@
...
+// };
+// alert(numberComparison(num1,num2));
+
+//Task 2.
+//Напишите функцию, которая принимает 2 параметра: длину и ширину прямоугольника и вычисляет его площадь. 
+//Если в функцию передали 1 параметр, то функция вычисляет площадь квадрата.
+//Подсказка: используйте параметры функции по умолчанию.
+//function squareArea(length, width = length) {
+//  const area = length * width;
+//  return area;
```
Comment (kathrynowy) [inline functions.js]:
```suggestion
//  return length * width;
```

File: functions.js [hunk 15]
```diff
@@ -0,0 +1,134 @@
...
+// console.log(getTrueArgument(0, 2, 5));
+ 
+//Task 6
+//Написать функцию, которая принимает 1 аргумент (кол-во итераций цикла)
+//и выводит на страницу столько красных кругов на страницу
+// const amountOfIterations = +prompt('Enter a number of iterations');
+
+// function redCircleDrawing(amountOfIterations) {
+//   if(isNaN(amountOfIterations)){
+//     alert('You entered a text. Enter a number of iterations')
```
Comment (kathrynowy) [inline functions.js]: add return here and remove unnecessary else below

so that your code will be more readable

File: functions.js [hunk 16]
```diff
@@ -0,0 +1,134 @@
...
+//   }else {
+//     for (let i = 1; i <= amountOfIterations; i++) {
+//       let div = document.createElement('div');
+//       div.className = 'red-circles';
+//       document.body.append(div);
+//     };
+//   };
+// };
```
Comment (kathrynowy) [inline functions.js]:
```suggestion
//   if(isNaN(amountOfIterations)){
//     alert('You entered a text. Enter a number of iterations')
return; 

//     for (let i = 1; i <= amountOfIterations; i++) {
//       let div = document.createElement('div');
//       div.className = 'red-circles';
//       document.body.append(div);
//     };
// };
```

Comment (kathrynowy) [reply functions.js]: you will notice it in VScode , that code is cleaner

File: functions.js [hunk 17]
```diff
@@ -0,0 +1,134 @@
...
+
+//Arguments property. Task 1.
+//Напишите функцию, которая принимает произвольное количество аргументов в виде строк
+//и выводит их через пробел с помощью document.write() в тело html-документа.
+//Например, функция  showWords("I've", "been", "learning", "JavaScript", "for", "a", "month")
+//должна вывести фразу "I've been learning JavaScript for a month".
+
+// function showWords() {
+//   let sentence = '';
+//   for (let i = 0; i < arguments.length; i++) {
```
Comment (kathrynowy) [inline functions.js]:
```suggestion
//   let sentence = '';

//   for (let i = 0; i < arguments.length; i++) {
```

File: functions.js [hunk 18]
```diff
@@ -0,0 +1,134 @@
...
+// showWords("I've", "been", "learning", "JavaScript", "for", "a", "month");
+
+
+//Task 2.
+//Напишите функцию, которая принимает произвольное количество аргументов в виде однозначных чисел и возвращает 1 многозначное число.
+//Например, функция с именем complexDigit() при вызове complexDigit(3,6,7) вернет число 367, а complexDigit(1,9, 4, 8,3) вернет число 19483.
+
+// function complexDigit() {
+//   let multivaluedNumber = '';
+//   for (let i = 0; i < arguments.length; i++) {
```
Comment (kathrynowy) [inline functions.js]:
```suggestion
//   let multivaluedNumber = '';

//   for (let i = 0; i < arguments.length; i++) {
```

File: index.html [hunk 19]
```diff
@@ -5,9 +5,12 @@
   <meta http-equiv="X-UA-Compatible" content="IE=edge">
   <meta name="viewport" content="width=device-width, initial-scale=1.0">
   <script src="./index.js"></script>
+  <script src="./cycles.js"></script>
+  <script src="./functions.js" defer></script>
```
Comment (kathrynowy) [inline index.html]: if you are working with DOM, where should you insert the script?? and why. answer in DM

**Review by kathrynowy — CHANGES_REQUESTED** [review]

File: functions.css (hunk 10 above)
Comment (ElenaLadutska) [reply functions.css]: fixed

File: functions.js (hunk 11 above)
Comment (ElenaLadutska) [reply functions.js]: fixed

File: functions.js (hunk 12 above)
Comment (ElenaLadutska) [reply functions.js:19]: okay, fixed

File: functions.js (hunk 13 above)
Comment (ElenaLadutska) [reply functions.js]: i forgot that function can return a numbers

File: functions.js (hunk 14 above)
Comment (ElenaLadutska) [reply functions.js]: fixed

File: functions.js (hunk 16 above)
Comment (ElenaLadutska) [reply functions.js]: Okay,  I fixed but I think after return should be }. without } it doesn't work

File: functions.js (hunk 17 above)
Comment (ElenaLadutska) [reply functions.js]: fixed

File: functions.js (hunk 18 above)
Comment (ElenaLadutska) [reply functions.js]: fixed

File: index.html (hunk 19 above)
Comment (ElenaLadutska) [reply index.html]: answered

File: functions.js [hunk 20]
```diff
@@ -8,43 +8,51 @@
 // function numberComparison(num1, num2) {
 //   if (num1 > num2) {
 //     return 1;
-//   }else if (num1 < num2) {
+
+//   } else if (num1 < num2) {
 //     return -1;
-//   }else if (isNaN(num1) || isNaN(num2)){
+
+//   } else if (isNaN(num1) || isNaN(num2)){
```
Comment (kathrynowy) [inline functions.js:15]: we check such cases at first, and only then we do logic

**Review by kathrynowy — APPROVED** [review]

**Pull request merged**